src/domains/dashboard/components/node-view/node-view.tsx

```diff
@@ -152,7 +154,9 @@ export const NodeView = ({
     ((((width * pcentWidth) / 100) * chartsMetadata.update_every) / 3) / 60,
   ) * 60
 
-  const menus = useMemo(() => renderChartsAndMenu(chartsMetadata), [chartsMetadata])
+  const menus = useMemo(() => renderChartsAndMenu(metricsCorrelationMetadata
+    || chartsMetadata, metricsCorrelationMetadata
+    && chartsMetadata), [metricsCorrelationMetadata, chartsMetadata])
```

Comment:
do you think that destructuring named arguments (ie. `renderChartsAndMenu({ chartsMetadata, metricsCorrelationMetadata })` would be more readable?

Reply:
Cleaned up a bit this part. Please retake a look